### PR TITLE
3560 add vim 8.2 and nvim 0.4 to ci tests

### PR DESCRIPTION
**Pull request by hsanson**
- Add vim 8.2 and nvim 0.4.4 to the CI script.
- Fix tests to work on vim 8.2 that has breaking changes.
- Update Docker image to use latest vim-vader.

Closes: #3560 and #3411

### CODE REVIEW

**Comment by hsanson** [comment]
@w0rp before this can be merged a new Docker image with the changes here must be build and pushed to Docker hub.

File: Dockerfile [hunk 1]
```diff
@@ -18,5 +20,8 @@ RUN apk --update add $PACKAGES && \
 
 RUN pip install vim-vint==0.3.15
 
-RUN git clone https://github.com/junegunn/vader.vim vader && \
-    cd vader && git checkout c6243dd81c98350df4dec608fa972df98fa2a3af
+RUN git clone https://github.com/junegunn/vader.vim vader
```
Comment (w0rp) [inline Dockerfile:23]: I recommend still targeting a specific commit or tag so it doesn't change later and break running some tests. Use the same commit ID in `.appveyor.yml` so the Windows tests run the same version.

File: Dockerfile [hunk 2]
```diff
@@ -2,8 +2,10 @@ FROM tweekmonster/vim-testbed:latest
 
 RUN install_vim -tag v8.0.0027 -build \
                 -tag v8.1.0519 -build \
+                -tag v8.2.2401 -build \
                 -tag neovim:v0.2.0 -build \
-                -tag neovim:v0.3.5 -build
+                -tag neovim:v0.3.5 -build \
+                -tag neovim:v0.4.4 -build
```
Comment (w0rp) [inline Dockerfile:8]: I think we can cut out the versions in the middle for Vim and NeoVim. Testing with the the oldest and newest supported versions should be enough to catch issues with functions being missing and so on.

**Review by w0rp — CHANGES_REQUESTED** [review]
I replied to your email about Docker access. I can add you to the dense-analysis organisation there, and then you should be able to set that up.

File: Dockerfile (hunk 2 above)
Comment (hsanson) [reply Dockerfile:8]: Got it, removed Vim 8.1 and Neovim 0.3.5

File: Dockerfile (hunk 1 above)
Comment (hsanson) [reply Dockerfile:23]: Done.  I am also trying to migrate appveyor to Github actions. At that point I may as well update vader on both sides.

**Comment by hsanson** [comment]
@w0rp I think this is ready to merge now. Some pointers:

 - The required checks must be updated in the settings afterwards.
 - The CI workflow will build and push the ALE docker image under these conditions:
    - The Dockerfile is modified.
    - The PR that modifies this file is not from a fork. This is a security limitation of Github Actions.

**Review by w0rp — APPROVED** [review]

**Comment by w0rp** [comment]
I've merged this and updated the branch protection rules to require the new checks. Great stuff.

Cheers! :beers:

**Pull request merged**